Makefile

```diff
@@ -63,20 +67,20 @@ scripts-permission:
 depends: $(BUILD_DIR)/ruby $(BUILD_DIR)/node
 
 # update dependencies information
-update: depends
+update: clean depends
 	./scripts/ruby bundle update
 	./scripts/node npm update
 
 # when we pull dependencies also pull docker image
 # without this images can get stale and out of sync from CI system
-$(BUILD_DIR)/ruby: Gemfile Gemfile.lock scripts/ruby | $(OUTPUT_DIR)
+$(BUILD_DIR)/ruby: Gemfile Gemfile.lock scripts/ruby vendor/gems | $(OUTPUT_DIR)
 	./scripts/ruby pull
 	./scripts/ruby bundle install --path=vendor/gems
 	@touch $(BUILD_DIR)/ruby
 
 # when we pull dependencies also pull docker image
 # without this images can get stale and out of sync from CI system
-$(BUILD_DIR)/node: package.json package-lock.json scripts/node | $(OUTPUT_DIR)
+$(BUILD_DIR)/node: package.json package-lock.json scripts/node node_modules | $(OUTPUT_DIR)
```

Comment:
If the dev deletes `node_modules`, the build should recover from that by reinstalling them.